src/nvim/os/env.c

```diff
@@ -427,6 +428,37 @@ char *vim_getenv(const char *name)
     return xstrdup(kos_env_path);
   }
 
+  // XDG Base Directory expansion.
+  if (strncmp(name, "XDG_", 4) == 0) {
+    const char *home = vim_getenv("HOME");
+    if (!home) {
+      return NULL;
+    }
+
+    const char *home_suffix = "";
+    if (strcmp(name, "XDG_DATA_HOME") == 0) {
+      home_suffix = "/.local/share/";
+    } else if (strcmp(name, "XDG_CONFIG_HOME") == 0) {
+      home_suffix = "/.config/";
+    } else if (strcmp(name, "XDG_CACHE_HOME") == 0) {
+      home_suffix = "/.cache/";
+    } else if (strcmp(name, "XDG_RUNTIME_DIR") == 0) {
+      return xstrdup("/tmp/");  // Arbitrary value, any idea?
+    } else if (strcmp(name, "XDG_CONFIG_DIRS") == 0) {
+      return xstrdup("/etc/xdg");
+    } else if (strcmp(name, "XDG_DATA_DIRS") == 0) {
+      return xstrdup("/usr/local/share/:/usr/share/");
+    }
+
+    // We need to append the prefix to the HOME path.
+    char *value = vim_getenv("HOME");
+    const size_t value_len = strlen(value);
+    const size_t home_suffix_len = strlen(home_suffix);
+    value = xrealloc(value, value_len + home_suffix_len + 1);
+    memmove(value + value_len, home_suffix, home_suffix_len + 1);
+    return value;
```

Comment:
Memory leak: `vim_getenv` returns an allocated value. `home` is neither used nor freed.